src/snowflake/snowpark/session.py

```diff
@@ -518,6 +546,7 @@ def _resolve_import_path(
         else:
             return trimmed_path, None, None
 
+    @Decorator.wrap_exception
     def _resolve_imports(
```

Comment:
I don't have strong opinion, but I wonder if we should just wrap it for public APIs if there is no information loss?

Reply:
My original thought is that we probably don't need to wrap "private" functions because customer should not be invoking them, but it is possible that these functions like `_resolve_imports` are used outside of this class.

Reply:
when I went through the code base to target the `ProgrammingError` exception I was specifically looking for any use of `_conn` and wrapped any functions using it. Given private functions in Python are not really private (a user may still access them!) I think it's probably best to wrap given the context, but no strong opinion on that.

Reply:
`wrap_exception` is already used by internal methods, for instance `ServerConnection.get_result_set`, which is called directly or indirectly by many public APIs. So we don't need to use `wrap_exception` on every public APIs.

Reply:
So I think this should be a bit more of a design decision. If we wrap at top-level public functions, then clearly we expose that any exception trace will be converted properly. If we have a mix of internal functions that may throw an exception together with a wrapper, and some functions wrapped/not wrapped we run the risk of some changes being introduced that may produce exceptions that are not translated properly. Therefore, I'd say let's remove all internal wrapping unless there's a very good reason and put the wrapper around any public interface. Thoughts?